### ISSUE TITLE
tf.compat.v1.profiler.profile produces wrong flops on Conv2D

### ISSUE THREAD

**Issue by SvenWarnke** [tensorflow/tensorflow#50802]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): Yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Windows 10
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: No
- TensorFlow installed from (source or binary): Pip
- TensorFlow version (use command below): 2.4.1
- Python version: 3.7.9
- Bazel version (if compiling from source): -
- GCC/Compiler version (if compiling from source): -
- CUDA/cuDNN version: - 
- GPU model and memory: -

Flops calculation for a Convolution2D Layer via tf.compat.v1.profiler.profile seems to be wrong. More specifically, it seems to only calculate flops for one application of the kernel and leave out the input size.

I use the following to calculate flops of a model containing only one conv layer:

`
from typing import Callable, Any
import tensorflow as tf

from tensorflow import keras
from tensorflow.keras import layers

def calculate_flops(model_generating_function: Callable, *args: Any, **kwargs: Any) -> int:
    """
    Calculates the flops of a keras model. For implementation reasons, which include creating and tearing down the
    tensorflow graph, the function does not take a model as input, but a function that returns said model.
    Adapted from answer of ch271828n on
    https://stackoverflow.com/questions/49525776/how-to-calculate-a-mobilenet-flops-in-keras.
    :param model_generating_function: a function that returns a keras model.
    :param args: positional arguments for model_generating_function
    :param kwargs: keyword arguments for model_generating_function
    :return: number of float point operations
    """
    tf.compat.v1.reset_default_graph()
    session = tf.compat.v1.Session()
    graph = tf.compat.v1.get_default_graph()

    with graph.as_default():
        with session.as_default():
            _ = model_generating_function(*args, **kwargs)

            run_meta = tf.compat.v1.RunMetadata()
            opts = tf.compat.v1.profiler.ProfileOptionBuilder.float_operation()

            # We use the Keras session graph in the call to the profiler.
            flops = tf.compat.v1.profiler.profile(
                graph=graph,
                run_meta=run_meta,
                cmd='op',
                options=opts
            )
    # necessary to not clog up graph and add flops over multiple calls to this function.
    tf.compat.v1.reset_default_graph()

    return flops.total_float_ops

def get_model(input_size: int, n_channels: int, n_filters: int, kernel_size: int) -> keras.Sequential:

    model = keras.Sequential(
        [
            keras.Input(shape=(input_size, input_size, n_channels)),
            layers.Conv2D(n_filters, kernel_size=(kernel_size, kernel_size), activation=None, padding="same", use_bias=False)
        ]
    )

    return model

assert calculate_flops(get_model, input_size=16, n_channels=1, n_filters=1, kernel_size=3) == 2 * 3 * 3 +1
`

will return 19 = 2 * kernel_size * kernel_size + 1

**Describe the expected behavior**
I would have expected something around 16 * 16 *  actual_output

**Comment by saikumarchalla** [comment]
@SvenWarnke  It looks like your Issue relates to the Keras component. Please submit it to the [github.com/keras-team/keras](github.com/keras-team/keras) repository instead. As ([previously announced](https://discuss.tensorflow.org/t/keras-project-moved-to-new-repository-in-https-github-com-keras-team-keras/1999/)), all future development of Keras is expected to happen in the [github.com/keras-team/keras](github.com/keras-team/keras) repository. If your issue lies with the TF-Core area please comment back with your explanation and we can look into it further. Thanks!

**Comment by saikumarchalla** [comment]
Closing due to lack of recent activity. Please update the issue when new information becomes available, and we will reopen the issue. Thanks!
